src/test/java/com/mentoring/test/FriendRequestTest.java

```diff
@@ -0,0 +1,104 @@
+import com.mentoring.framework.database.Database;
+import com.mentoring.framework.database.DbSelects;
+import com.mentoring.framework.utils.UserUtils;
+import com.mentoring.model.User;
+import com.mentoring.pageobject.*;
+import com.mentoring.test.assertionsteps.CommonAssertions;
+import com.mentoring.util.Messages;
+import io.qameta.allure.Feature;
+import io.qameta.allure.Severity;
+import io.qameta.allure.SeverityLevel;
+import lombok.extern.slf4j.Slf4j;
+import org.assertj.core.api.SoftAssertions;
+import org.testng.annotations.*;
+
+import java.sql.SQLException;
+
+import static com.mentoring.model.Features.FRIEND_REQUEST;
+
+@Slf4j
+public class FriendRequestTest extends BasicTest {
+
+    private SoftAssertions softAssertion;
+    private Database database;
+    private User userA;
+    private User userB;
+    private String characterNameA;
+    private String characterNameB;
+    String accessTokenIdForUserA;
+    String accessTokenIdForUserB;
+    String userIdForUserA;
+    String userIdForUserB;
+
+    @BeforeClass(alwaysRun = true)
+    private void setup() {
+        database = new Database();
+    }
+
+    @AfterClass(alwaysRun = true)
+    private void close() {
+        database.disconnectFromDb();
+    }
+
+    @BeforeMethod(alwaysRun = true)
+    private void initMethod() throws SQLException {
+        softAssertion = new SoftAssertions();
+
+        // create userA, login and create a character for it with RestAPI
+        userA = new User();
+        userA.loginUser(userA.getUserName(), userA.getPassword());
+        accessTokenIdForUserA = DbSelects.getAccessTokenIdByEmailAddress(database, userA.getEmail());
+        userIdForUserA = DbSelects.getUserIdByEmailAddress(database, userA.getEmail());
+        characterNameA = UserUtils.generateRandomCharacterName();
+        userA.createCharacter(characterNameA, accessTokenIdForUserA, userIdForUserA);
+
+        // create userB, login and create a character for it with RestAPI
+        userB = new User();
+        userB.loginUser(userB.getUserName(), userA.getPassword());
+        accessTokenIdForUserB = DbSelects.getAccessTokenIdByEmailAddress(database, userB.getEmail());
+        userIdForUserB = DbSelects.getUserIdByEmailAddress(database, userB.getEmail());
+        characterNameB = UserUtils.generateRandomCharacterName();
+        userB.createCharacter(characterNameB, accessTokenIdForUserB, userIdForUserB);
+    }
+
+    @AfterMethod(alwaysRun = true)
+    private void terminate() {
+        userA.logoutUser(userA.getUserName(), accessTokenIdForUserA, userIdForUserA);
+        userB.logoutUser(userB.getUserName(), accessTokenIdForUserB, userIdForUserB);
+    }
+
+    @Test(groups = "smoke")
+    @Severity(SeverityLevel.BLOCKER)
+    @Feature(FRIEND_REQUEST)
+    public void sendAndReceiveFriendRequest() {
+        log.info("usernameA = {}; characterNameA = {}; userNameB = {}; characterNameB = {}", userA.getUserName(), characterNameA, userB.getUserName(), characterNameB);
```

Comment:
I wouldn't put logs to test level. Logs rather belong to page objects and utility classes.